test/integration/helpers/integration-test-editor.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import {
+	BlockEditorKeyboardShortcuts,
+	BlockEditorProvider,
+	BlockList,
+	BlockTools,
+	BlockInspector,
+	WritingFlow,
+	ObserveTyping,
+} from '@wordpress/block-editor';
+import { Popover, SlotFillProvider } from '@wordpress/components';
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { ShortcutProvider } from '@wordpress/keyboard-shortcuts';
+import '@wordpress/format-library';
+import {
+	createBlock,
+	unregisterBlockType,
+	getBlockTypes,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { waitForStoreResolvers } from './wait-for-store-resolvers';
+
+// Polyfill for String.prototype.replaceAll until CI is runnig Node 15 or higher.
+if ( ! String.prototype.replaceAll ) {
+	String.prototype.replaceAll = function ( str, newStr ) {
+		// If a regex pattern
+		if (
+			Object.prototype.toString.call( str ).toLowerCase() ===
+			'[object regexp]'
+		) {
+			return this.replace( str, newStr );
+		}
+
+		// If a string
+		return this.replace( new RegExp( str, 'g' ), newStr );
+	};
+}
+
+/**
+ * For registering all the core blocks if needed as part of test setup.
+ */
+export function registerAllCoreBlocks() {
+	registerCoreBlocks();
```

Comment:
I'm wondering if we could do this in `initializeEditor` automatically?

Reply:
Yeh, I did start with that, but was thinking that at some point we might want to export this test editor setup from a package so that the likes of woo, and other 3rd party blocks could use it, and they may not want the core blocks loading.

Reply:
@kevin940726 I took another look at this and moved the block setup into  `initializeEditor` and the teardown into an effect return in the editor, with a param to disable the creation of the core blocks if not needed. This will remove a bit of boiler plate from each of the test suites.

What do you think of this approach?

Reply:
@getdave you may want to take look at this change as well.